bot/admin/admin.py

```diff
@@ -258,6 +261,101 @@ async def change_discord_presence_clear(self, ctx: commands.Context):
+        """Command handler for the `botonly` command.
+
+        This command marks a channel in the database as bot-only, so every message posted by someone else than the bot
+        will be deleted immediately.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context from which this command is invoked.
+            channel (discord.Textchannel): The channel that is to be made bot-only
+        """
+        is_channel_botonly = self._db_connector.is_botonly(channel)
+        if is_channel_botonly:
+            log.info("Deactivated botonly for channel {0}".format(channel))
+            self._db_connector.disable_botonly(channel)
+        else:
+            log.info("Activated botonly for channel {0}".format(channel))
+            self._db_connector.enable_botonly(channel)
+
+        is_enabled_string = 'aktiviert' if not is_channel_botonly else 'deaktiviert'
+        embed = _build_botonly_embed(is_enabled_string)
+        await channel.send(embed=embed)
+        message = await ctx.send("Bot-Only wurde für den Channel {0} {1}".format(channel.mention, is_enabled_string))
+        await message.delete(delay=60.0)
+
+    @commands.command(name="purge")
+    @command_log
+    async def purge_channel(self, ctx: commands.Context, channel: discord.TextChannel):
+        """Command handler for the `purge` command.
+
+        Removes all messages in a channel.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context from which this command is invoked.
+            channel (discord.Textchannel): The channel that is to purge
+        """
+        embed = _build_purge_confirmation_embed(channel)
+        timeout = 15.0
+        reaction = await self._send_confirmation_dialog(ctx, embed, timeout)
+        if reaction is None:
+            return
+        if str(reaction[0].emoji) == constants.EMOJI_CONFIRM:
+            await _purge_channel(channel)
+
+    @commands.Cog.listener(name='on_message')
+    async def on_message(self, ctx: discord.Message):
+        """Event Handler for new messages.
+
+        Deletes a message if the channel it was posted in is a botonly channel and the message was not by a bot
+
+        Args:
+            ctx (discord.Message): The context this method was called in. Must always be a message.
+        """
+        if ctx.author == self.bot.user:
+            return
+        if self._db_connector.is_botonly(ctx.channel):
+            await ctx.delete()
+
+    async def _send_confirmation_dialog(self, ctx: commands.Context, embed: discord.Embed, timeout: float) -> \
+            Optional[Tuple[discord.Reaction, discord.User]]:
+        """Handles a confirmation dialog and returns the user reaction.
+
+        Prints a passed embed and adds reactions for confirmation and cancellation. The reaction that is clicked will be
+        returned to the user. If no reaction is clicked before a timeout None is returned instead.
```

Comment:
.. will be returned.  If no reaction has been clicked before the timeout is reached, `None` is returned instead.